tune_sklearn/tune_search.py

```diff
@@ -406,17 +411,21 @@ def __init__(
                     elif scheduler == "MedianStoppingRule":
                         self.scheduler = MedianStoppingRule(
                             metric="average_test_score")
+                    elif scheduler == "ASHAScheduler":
+                        self.scheduler = ASHAScheduler(
+                            metric="average_test_score")
                 else:
                     raise ValueError("{} is not a defined scheduler. "
                                      "Check the list of available schedulers."
                                      .format(scheduler))
-            elif isinstance(scheduler, TrialScheduler) or scheduler is None:
+            elif isinstance(scheduler, TrialScheduler):
                 self.scheduler = scheduler
-                if self.scheduler is not None:
-                    self.scheduler.metric = "average_test_score"
+                self.scheduler.metric = "average_test_score"
+            elif scheduler is None:
+                self.scheduler = ASHAScheduler(metric="average_test_score")
             else:
                 raise TypeError("Scheduler must be a str or tune scheduler")
-        else:
+        elif self.early_stopping:
             warnings.warn("Unable to do early stopping because "
                           "estimator does not have `partial_fit`")
             self.early_stopping_max_epochs = 1
```

Comment:
Add `else` case to set `self.scheduler` to something if none of these if cases are satisfied